### ISSUE TITLE
badrequestobjectresult.value doesn't recognized by pattern type of serializableError

### ISSUE THREAD

**Issue by LittleKing2020** [dotnet/docs#33794]
badrequestobjectresult.value doesn't recognized by pattern type of serializableError
`List<string> message = new List<string>();
                if (badRequestObjectResult.Value is SerializableError errors) 
                {
                    var errorMessages = errors.SelectMany(p => (string[])p.Value).Distinct();
                    message.AddRange(errorMessages);
                }
                else
                    message.Add(badRequestObjectResult.Value.ToString());`
And It returns  Microsoft.AspNetCore.Mvc.ValidationProblemDetails

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: bbb051ee-f020-66ba-c601-4a08ad32ef61
* Version Independent ID: 58a19505-ac46-c5c8-12cc-fe87b6aa18a7
* Content: [Sorry, we don't have specifics on this C# error](https://learn.microsoft.com/en-us/dotnet/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error?f1url=%3FappId%3Droslyn%26k%3Dk(CS8121))
* Content Source: [docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md](https://github.com/dotnet/docs/blob/main/docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md)
* Product: **dotnet-csharp**
* Technology: **csharp-diagnostics**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Hi @LittleKing2020 

We're working to add more pages for these errors and warnings.

In the meantime, to help you get unblocked, I suggest posting the question with a larger code snippet in the [C# section of our Q & A site](https://learn.microsoft.com/answers/tags/309/csharp).

I'll close this as it's already tracked in other work items.
